### PR TITLE
Add page-level metadata to units and modules

### PR DESCRIPTION
**Pull request by jamarw**
1. Add page-level metadata to units and modules
2. Use **gitHubID** and **alias** settings to populate author and ms.author values if available. If a user hasn't set values, use default placeholder (this will result in build warnings as a reminder to update values).
3. Add content property to units and prerequisites to modules.

### CODE REVIEW

**Comment by jamarw** [comment]
1. Add page-level metadata to units and modules
2. Use **gitHubID** and **alias** settings to populate author and ms.author values if available.  If a user hasn't set values, use default placeholder (this will result in build warnings as a reminder to update values).
3. Add content property to units and prerequisites to modules.

File: docs-article-templates/src/helper/unit-builder.ts [hunk 1]
```diff
@@ -64,17 +84,20 @@ export async function cleanupUnit(generatedUnit: string) {
     try {
         const moduleContent = readFileSync(generatedUnit, "utf8");
         const updatedModule = moduleContent.replace("header: ", "")
-            .replace(/'/g, "");
+            .replace(/'/g, "")
+            .replace(`content: |-`, "content: |")
+            .replace(/^\s*[\r\n]/gm, "");
         writeFileSync(generatedUnit, updatedModule, "utf8");
```
Comment (bharney-msft) [inline docs-article-templates/src/helper/unit-builder.ts]: Is it possible to use `writeFile` or does it need to be `writeFileSync`?

File: docs-article-templates/src/helper/unit-builder.ts [hunk 2]
```diff
@@ -41,16 +43,34 @@ export async function createUnits() {
...
     }
+    if (!gitHubID) {
+        author = `...`;
+    }
+    if (!alias) {
+        msAuthor = `...`;
+    }
+
     /* tslint:disable:object-literal-sort-keys */
     const yaml = require("write-yaml");
```
Comment (bharney-msft) [inline docs-article-templates/src/helper/unit-builder.ts]: Have you tried to use `js-yaml` Im pretty sure `write-yml` uses `js-yaml.dump(obj)` under the hood. Its a more widely used npm package.

Comment (jamarw) [reply docs-article-templates/src/helper/unit-builder.ts]: Hi @bharney-msft , good idea.  Updated.

**Pull request merged**